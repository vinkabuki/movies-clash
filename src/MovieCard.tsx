import React from 'react'
import styled from 'styled-components'
import { useSwipeable } from 'react-swipeable'
const ItemBox = styled.div`
  background-color: blue;
  max-width: 500px;
  margin: auto;
  padding: 30px;
`
const CoverImage = styled.img`
  width: 100%;
`
const MovieTitle = styled.h1`
  text-align: left;
`
const Summary = styled.p`
  text-align: left;
`
const MainButton = styled.button`
  width: 50%;
  background-color: ${(props) => props.bgcolor};
`

interface IProps {
  item: {
    id: string
    imageURL: string
    title: string
    summary: string
    rating: string
  }
  onAcceptHandler: any
  onRejectHandler: any
  //handler: any
}

export const MovieCard = ({ item, onAcceptHandler, onRejectHandler }: IProps) => {
  const handler = useSwipeable({
    onSwiped: (id) => {
      onRejectHandler(id)
    }
  })

  return (
    <ItemBox {...handler}>
      <CoverImage src={item.imageURL} alt="there is no cover image for this movie, sorry" />
      <MovieTitle>{item.title}</MovieTitle>
      <Summary>{item.summary}</Summary>
      <h2>{item.rating}/10</h2>
      <MainButton onClick={() => onRejectHandler(item.id)} bgcolor="red">
        Reject
      </MainButton>
      <MainButton onClick={() => onAcceptHandler(item.id)} bgcolor="green">
        Accept
      </MainButton>
    </ItemBox>
  )
}

export default MovieCard
