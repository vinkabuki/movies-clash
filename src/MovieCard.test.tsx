import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import MovieCard from './MovieCard'

describe('MoviesCards component basic render test', () => {
  const currentItem = {
    id: 'cool99',
    imageURL: 'https://google.com/images.jpg',
    title: 'Matrix',
    summary: 'lorem ipsum',
    rating: 6.7
  }
  test('Render example movie item', () => {
    const component = renderer.create(<MovieCard item={currentItem} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
