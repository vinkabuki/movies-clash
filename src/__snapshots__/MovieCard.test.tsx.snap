// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesCards component basic render test Render example movie item 1`] = `
.c0 {
  position: relative;
  margin: auto;
  border-radius: 10px;
  overflow: hidden;
  background-image: url(https://google.com/images.jpg);
  background-size: 100%;
  background-repeat: no-repeat;
  min-height: 580px;
  box-shadow: 10px 7px 65px -13px rgba(0,0,0,0.75);
}

.c1 {
  padding: 20px;
  position: absolute;
  bottom: 20px;
}

.c7 {
  box-sizing: border-box;
  padding: 40px;
  max-width: 400px;
  width: 100%;
  margin: auto;
  margin-top: 590px;
  display: none;
}

.c2 {
  text-align: left;
  color: white;
}

.c5 {
  text-align: left;
  color: white;
}

.c4 {
  text-align: left;
  color: white;
}

.c3 {
  background-color: rgba(0,0,0,0.3);
}

.c6 {
  width: 30px;
  height: 30px;
  position: absolute;
  right: 10px;
  bottom: 10px;
  border-radius: 90px;
  cursor: pointer;
  border: none;
  font-size: 14px;
  outline: 0;
  box-shadow: 0 0 0 0 rgba(0,0,0,1);
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-animation: pulse 2s infinite;
  animation: pulse 2s infinite;
}

@media (max-width:400px) {
  .c0 {
    min-height: calc(100vw * 1.45);
  }
}

@media (max-width:400px) {
  .c7 {
    margin-top: calc(100vw * 1.45);
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h1
      className="c2"
    >
      <span
        className="c3"
      >
        Matrix
      </span>
    </h1>
    <p
      className="c4"
    >
      <span
        className="c3"
      >
        6.7
        /10
      </span>
    </p>
    <p
      className="c5"
    >
      <span
        className="c3"
      >
        lorem ipsum
      </span>
    </p>
  </div>
  <button
    className="c6"
    onClick={[Function]}
  >
    i
  </button>
  <div
    className="c7"
  >
    <h1
      className="c2"
    >
      Matrix
    </h1>
    <p
      className="c4"
    >
      6.7
      /10
    </p>
    <p
      className="c5"
    >
      lorem ipsum
    </p>
  </div>
</div>
`;
